libraries/AP_Compass/AP_Compass_HMC5843.cpp

```diff
@@ -316,22 +332,13 @@ bool AP_Compass_HMC5843::_read_sample()
     } val;
     int16_t rx, ry, rz;
 
-    if (_retry_time > AP_HAL::millis()) {
-        return false;
-    }
-
     if (!_bus->block_read(HMC5843_REG_DATA_OUTPUT_X_MSB, (uint8_t *) &val, sizeof(val))){
-        _retry_time = AP_HAL::millis() + 1000;
         return false;
     }
 
     rx = be16toh(val.rx);
-    ry = be16toh(val.ry);
-    rz = be16toh(val.rz);
-
-    if (_is_hmc5883L) {
-        std::swap(ry, rz);
-    }
+    ry = be16toh(val.rz);
+    rz = be16toh(val.ry);
```

Comment:
since now we don't support HMC5843 anymore, it would be good to add a comment here. Otherwise it will trigger incorrect fixes of "hey, this is assigning the wrong axis while doing the endianess conversion".

Renaming the driver would be a good thing, too.